### PR TITLE
add psi and cpv documentation

### PR DESCRIPTION
**Pull request by Piranias**
Fix #Issue

**Changes proposed in this pull request**:
- add documentation for CPV and PEROSI technologies

The following steps were realized, as well (if applies):
- [ ] Use in-line comments to explain your code
- [ ] Write docstrings to your code
- [ ] For new functionalities: Explain in readthedocs
- [ ] Write test(s) for your new patch of code
- :x: Update the CHANGELOG.md (let's start this after the first release)
- [ ] Apply black (`black . --exclude docs/`)

*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

:x: Check not applicable to this PR

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/mvs_eland/blob/dev/CONTRIBUTING.md).*<sub>

### CODE REVIEW

**Comment by Piranias** [comment]
one function does not come our right. There is a bug 

`This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019/Debian) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
`
locally it is displayed correctly, even though it already shows these errors also locally.. any idea?

**Comment by Piranias** [comment]
@SabineHaas There is still one function that does not dislay correctly. 
see https://pvcompare.readthedocs.io/en/documentation/pv_feedin.html

I have no idea how to change this. I already added some extensions to the config.py but it didnt help. Do you have any idea?

File: docs/pv_feedin.rst [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+While the SI module feedin timeseries is completely calculated with `pvlib <https://pvlib-python.readthedocs.io/en/stable/index.html>`_ ,
+unique models were developed for the CPV and PeroSi technologies. The next
+sections will provide a detailed descriptions of the different modeling
+approaches.
+
+------------------
+1. SI
+------------------
+The silicone module parameters are loaded from cec module database. The module
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:20]: could you add a link please?

File: docs/pv_feedin.rst [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+selected by default is the "Aleo_Solar_S59y280" module with a 17% efficiency.
+But any other module can be selected.
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:22]: Do you have to hard-code it or how can you select it? If hard-coded, please write an issue, to change this later.

File: docs/pv_feedin.rst [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+The timeseries is calculating making usage of the `Modelchain  <https://pvlib-python.readthedocs.io/en/stable/modelchain.html>`_
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:24]:
```suggestion
The time series is calculated by making usage of the `Modelchain  <https://pvlib-python.readthedocs.io/en/stable/modelchain.html>`_
```

File: docs/pv_feedin.rst [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+functionality in `pvlib <https://pvlib-python.readthedocs.io/en/stable/index.html>`_. In order to make the results compareable for real world
+conditions the following methods are added to the `modelchain object <https://pvlib-python.readthedocs.io/en/stable/api.html#modelchain>`_ :
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:26]: are "selected"? I think the methods already exist and you select them by setting parameters in the modelchain.

File: docs/pv_feedin.rst [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+- spectral_model="first_solar"
+- temperature_model="sapm"
+- losses_model="pvwatts"
+
+
+-------
+2. CPV
+-------
+
+The CPV technology that is used in the pvcopare simulations is a hybrid
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:38]:
```suggestion
The CPV technology that is used in the pvcompare simulations is a hybrid
```

File: docs/pv_feedin.rst [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+refractive, diffuse light which is not focused onto the III-V cells
+is instead collected by the Si cells, which cover the area not
+taken up by III-V cells. Voltages are not matched between III-
+V and Si cells, so a four terminal output is provided." (From `Askins 2019 <https://zenodo.org/record/3349781#.X46UFZpCT0o>`_)
+
+Modeling the hybrid system
+--------------------------
+The model of the cpv technology is outsourced from pvcompare and can be found in the
+`cpvlib <https://github.com/isi-ies-group/cpvlib>`_ repository. PVcompare
+contains the wrapper function `apply_cpvlib_StaticHybridSystem`.
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:70]: please add a link to the function

File: docs/pv_feedin.rst [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+In order to model the dependencies of AOI, temperature and spectrum of the cpv
+module, the model follows an approach of `[Gerstmeier, 2011] <https://www.researchgate.net/publication/234976094_Validation_of_the_PVSyst_Performance_Model_for_the_Concentrix_CPV_Technology>`_
+Previously implemented for CPV in PVSYST. The approach uses the single diode
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:74]:
```suggestion
previously implemented for CPV in PVSYST. The approach uses the single diode
```

File: docs/pv_feedin.rst [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+The overall model for the hybrid system is illustrated in the next figure.
+
+.. image:: ./images/StaticHybridSystem_block_diagram.png
+  :width: 100%
+  :alt: modeling scheme of the hybrid micro-concentrator module
+
+CPV submodule
+-------------
+
+Input parameters are weather data with AM (airmass), Temoerature,
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:107]:
```suggestion
Input parameters are weather data with AM (airmass), temperature,
```

File: docs/pv_feedin.rst [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+The CPV part only takes DNI into account. The angle of incidence (AOI) is calculated
+by `pvlib.irradiance.aoi() <https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.irradiance.aoi.html?highlight=pvlib.irradiance.aoi#pvlib.irradiance.aoi>`_.
+Further the `pvlib.pvsystem.singlediode() <https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.pvsystem.singlediode.html?highlight=singlediode>`_ function is solved for the given module parameters.
+The utilization factors have been defined before by correlation analysis of
+outdoor measurements. The given utilization factors for temperature and airmass
+are then multiplied with the output power of the single diode functions. They
+function as temperature and airmass corrections due to spectral and temperature
+losses.
+
+flatplate submodule
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:118]:
```suggestion
Flatplate submodule
```

File: docs/pv_feedin.rst [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+-------------------
+
+For AOI < 60° the flatplate part only takes GII (global inclined irradiance) -
+DII (direct inclined irradiance). So only the diffuse part of the irradiance
+is considered. For Aoi > 60 ° GII is considered because DII and DHI come through.
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:123]: Maybe this way it's easier to understand:
For AOI < 60° only the diffuse irradiance reaches the flate plate module, which is why ... : GII (global inclined irradiance) - DII (direct inclined irradiance). 
For Aoi > 60 ° also DII and DHI fall onto the flate plate module, which is why...

File: docs/pv_feedin.rst [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+The single diode equation is then solved for all timesteps with the specific
+input irradiance. No module connection is assumed, so cpv and flatplate output
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:125]:
```suggestion
input irradiance. No module connection is assumed, so CPV and flat plate output
```

File: docs/pv_feedin.rst [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+week measurement in Madrid in May 2019. The Data can be found here
+`UPM measurement data <https://zenodo.org/record/3346823#.X46UDZpCT0o>`_ ,
+whereas the paper discribing the performance testing of the test module can
+be found here `Askins_2019 <https://zenodo.org/record/3349781#.X46UFZpCT0o>`_
+
+
+------------------
+2. PeroSi
+------------------
+The perovskite silicone cell is a high efficiency cell that is still in it's
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:141]:
```suggestion
The perovskite-silicon cell is a high-efficiency cell that is still in its
```

File: docs/pv_feedin.rst [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+test phase. Because perovskite is a material that is easily accessible many
+researchers around the world are investigating the potential of perovskite and
+tandem e.g perovskite - silicone cells, which we will focus on here.
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:144]: of perovskite in tandem cells?

File: docs/pv_feedin.rst [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+Because of the early stage of the
+development of the technology, no outdoor measurement data is available to
+draw correlations for temperature dependencies or spectral dependencies which
+are of great impact for multijunction cells.
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:148]:
```suggestion
are of great impact for multi-junction cells.
```

File: docs/pv_feedin.rst [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+(a Simple Model of the Atmospheric Radiative Transfer of Sunshine) to the given
+EQE curves of our model. Temperature dependencies are covered by a temperature
+coefficient for each sub cell. The dependence of AOI is taken into account
+by `SMARTS <https://www.nrel.gov/grid/solar-resource/smarts.html>`_.
+
+.. image:: ./images/schema_modell.jpg
+  :width: 100%
+  :alt: modeling scheme of the perovskite silicone tandem cell
+
+input data
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:165]:
```suggestion
Input data
```

File: docs/pv_feedin.rst [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+* cell parameters for each subcell:
+    * series resistance (Rs)
+    * shunt resistance (R_shunt)
+    * saturation current (j_0)
+    * Temperature coefficient for the short circuit current (α)
+    * Energy Bandgap
+    * cell size
+    * external quantum efficiency curve (EQE-curve)
+
+The cell parameters provided in pvcompare are for the cells (Korte...) ith 17 %
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:180]: reference is missing

File: docs/pv_feedin.rst [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+efficiency and (Chen) bin 29% efficiency. For Chen the parameters Rs, R_shunt
+and j_0 are found by fitting the IV curve.
+
+modeling procedure
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:184]:
```suggestion
Modeling procedure
```

File: docs/pv_feedin.rst [hunk 18]
```diff
@@ -0,0 +1,248 @@
...
+------------------
+1. **weather data**
+The POA_global (plane of array) irradiance is calculated with `pvlib.irradiance.get_total_irradiance() <https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.irradiance.get_total_irradiance.html#pvlib.irradiance.get_total_irradiance>`_ function
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:187]:
```suggestion
The POA_global (plane of array) irradiance is calculated with the `pvlib.irradiance.get_total_irradiance() <https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.irradiance.get_total_irradiance.html#pvlib.irradiance.get_total_irradiance>`_ function
```

File: docs/pv_feedin.rst [hunk 19]
```diff
@@ -0,0 +1,248 @@
...
+
+    \text q : \text{elementary electric charge}
+
+3. The `pvlib.pvsystem.singlediode() <https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.pvsystem.singlediode.html?highlight=singlediode>`_
+function is used to evaluate the output power of each
+subcell.
+
+3.1 The output power Pmp is multiplied by the number of cells in series
+
+3.2 losses due to cell connection (5%) and cell to module connection (5%) are
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:212]:
```suggestion
3.2 Losses due to cell connection (5%) and cell to module connection (5%) are
```

File: docs/pv_feedin.rst [hunk 20]
```diff
@@ -0,0 +1,248 @@
...
+taken into account
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:213]:
```suggestion
taken into account.
```

File: docs/pv_feedin.rst [hunk 21]
```diff
@@ -0,0 +1,248 @@
...
+
+
+2) **Normalize by p_mp at standard test conditions (power at maximum powerpoint) (NSTC)**
+
+* This procedure accounts for all losses under real world conditions and displays the difference between ideal operation and real world operation
+
+
+3) **Normalize by p_mp at real world conditions (NRWC)**
+
+* This procedure calculates the maximum powerpoint for realorld conditions at irr_ref = 1000 W/qm and temp_ref = 25 °.
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:245]:
```suggestion
* This procedure calculates the maximum power point for real world conditions at irr_ref = 1000 W/qm and temp_ref = 25°C.
```

File: pvcompare/cpv/apply_cpvlib_StaticHybridSystem.py [hunk 22]
```diff
@@ -7,20 +7,29 @@
 
 
 def create_cpv_time_series(lat, lon, weather, surface_azimuth, surface_tilt):
+
     """
-    creates a time series for a type of cpv module
+    creates a time series for a cpv module.
+
+    notice: right now only module parameters for the hybrid INSOLIGHT module is
+    provided. If you want to add you own module data please do this in
```
Comment (SabineHaas) [inline pvcompare/cpv/apply_cpvlib_StaticHybridSystem.py:15]: Something is missing in the end

File: pvcompare/perosi/pvlib_smarts.py [hunk 23]
```diff
@@ -20,84 +20,33 @@ def SMARTSSpectra(
...
-            36* Direct normal photosynthetic photon flux ?mol m-2 s-1 nm-1
-            37* Diffuse horizontal photosynthetic photon flux ?mol m-2 s-1 nm-1
-            38* Global tilted photosynthetic photon flux ?mol m-2 s-1 nm-1
-            39* Spectral photonic energy eV
-            40* Global horizontal photon flux per eV cm-2 s-1 eV-1
-            41* Direct normal photon flux per eV cm-2 s-1 eV-1
-            42* Diffuse horizontal photon flux per eV cm-2 s-1 eV-1
-            43* Global tilted photon flux per eV cm-2 s-1 eV-1
+    IOUT: str
+        all output variables can be found in the SMARTS 2.9.5 documentation
```
Comment (SabineHaas) [inline pvcompare/perosi/pvlib_smarts.py:30]: can you provide a link?

File: pvcompare/pv_feedin.py [hunk 24]
```diff
@@ -446,7 +446,11 @@ def create_cpv_time_series(
     Returns
     -------
     :pandas:`pandas.Series<series>`
-        Power output of CPV module in W.
+
+        Power output of CPV module in W (if parameter `normalized` is False) or
+        normalized power output of CPV module (if parameter `normalized` is
+        False).
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:452]: I think the parameter is called `normalization` now.

**Review by SabineHaas — CHANGES_REQUESTED** [review]
Nice work, and a lot of information :) 

General comments:
- We should decide on "pvcompare" or "PVcompare". I would say "pvcompare" and write it kursive in the docs.
- Did you ask for permission of using the figures?  If not, I can ask.
- Could you add references like in MVS please? With a reference section at the end of the page.
- figure captions are not displayed

File: docs/pv_feedin.rst [hunk 25]
```diff
@@ -0,0 +1,248 @@
...
+    \left\{
+    \begin{array}{
+        @{}% no padding
+        l@{\quad}% some padding
+        r@{}% no padding
+        >{{}}r@{}% no padding
+        >{{}}l@{}% no padding
+    }
+        1 + (x - x_{thrd}) \cdot S_{x\leq x_{thrd}}&         & \text{if }   x &\leq x_{thrd} \\
+        1 + (x - x_{thrd}) \cdot S_{x\leq x_{thrd}}&         & \text{if }    x &\geq x_{thrd}
```
Comment (SabineHaas) [inline docs/pv_feedin.rst:94]: Above and below the same before if ? (sorry if I'm to tired to find the difference :D)

File: pvcompare/pv_feedin.py (hunk 24 above)
Comment (Piranias) [reply pvcompare/pv_feedin.py:452]: true!

Comment (Piranias) [reply pvcompare/pv_feedin.py:452]: this has been changed in dev already

File: pvcompare/perosi/pvlib_smarts.py (hunk 23 above)
Comment (Piranias) [reply pvcompare/perosi/pvlib_smarts.py:30]: >     * Could you add references like in MVS please? With a reference section at the end of the page.

What do you mean exactly? I didn't see any reference section at the end of the page in mvs. can you give me an example link?

**Comment by Piranias** [comment]
* Could you add references like in MVS please? With a reference section at the end of the page.

What do you mean exactly? I didn't see any reference section at the end of the page in mvs. can you give me an example link?

File: pvcompare/perosi/pvlib_smarts.py (hunk 23 above)
Comment (Piranias) [reply pvcompare/perosi/pvlib_smarts.py:30]: done

File: pvcompare/cpv/apply_cpvlib_StaticHybridSystem.py (hunk 22 above)
Comment (Piranias) [reply pvcompare/cpv/apply_cpvlib_StaticHybridSystem.py]: done

File: docs/pv_feedin.rst (hunk 25 above)
Comment (Piranias) [reply docs/pv_feedin.rst]: done

**Review by SabineHaas — APPROVED** [review]
Approved after the following changes:
- display caption of second figure https://pvcompare.readthedocs.io/en/documentation/pv_feedin.html

**Pull request merged**